sunbeam-python/sunbeam/steps/maintenance.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from abc import ABC, abstractmethod
+
+from rich.console import Console
+from rich.status import Status
+from watcherclient import v1 as watcher
+from watcherclient.v1 import client as watcher_client
+
+from sunbeam.core import watcher as watcher_helper
+from sunbeam.core.common import BaseStep, Result, ResultType, SunbeamException
+from sunbeam.core.deployment import Deployment
+from sunbeam.core.questions import ConfirmQuestion
+
+LOG = logging.getLogger(__name__)
+
+
+class WatcherStepMixin(ABC, BaseStep):
+    name: str = ""
+    description: str = ""
+
+    def __init__(
+        self,
+        deployment: Deployment,
+    ):
+        super().__init__(self.name, self.description)
+        self.client: watcher_client.Client = watcher_helper.get_watcher_client(
+            deployment=deployment
+        )
+
+        self.confirm: bool | None = False
+        self.audit: watcher.Audit
+
+    def has_prompts(self) -> bool:
+        """Returns true if the step has prompts that it can ask the user.
+
+        :return: True if the step can ask the user for prompts,
+                 False otherwise
+        """
+        return True
+
+    def is_skip(self, status: Status | None = None) -> Result:
+        """Determines if the step should be skipped or not.
+
+        :return: ResultType.SKIPPED if the Step should be skipped,
+                 ResultType.COMPLETED or ResultType.FAILED otherwise
+        """
+        if self.confirm:
+            return Result(ResultType.COMPLETED)
+        return Result(ResultType.SKIPPED)
+
+    def _confirm_audit(self, audit) -> bool | None:
+        """Get actions created by audit and get confirm from user."""
+        actions = watcher_helper.get_actions(client=self.client, audit=audit)
+        action_operation_msg = ""
+        if len(actions) == 0:
+            return True
+        for idx, action in enumerate(actions):
+            action_operation_msg += (
+                f"\t{idx}. {action.description}"
+                f" | parameters: {action.input_parameters}\n"
+            )
+        question = ConfirmQuestion(
+            ("Confirm to run operations for cluster:\n" f"{action_operation_msg}"),
```

Comment:
Can you update in Github Conversation first commit message box with the UX along with output.

Is the user will be prompted to accept the actions for every Step that contains has_prompts as True? In other words, user accepts the actions for Hypervisor maintenance using watcher once and then after some time for Storage related?

Do you have plan to add maintenance commands that list all actions required across hypervisor/k8s/microceph?
Do you have plan to add flag to proceed with maintenance without prompts once user has reviewed the actionplan in total?